src/ServiceControl/Infrastructure/Settings/Settings.cs

```diff
@@ -433,5 +433,19 @@ private static RemoteInstanceSetting[] GetRemoteInstances()
             }
             return new RemoteInstanceSetting[0];
         }
+
+        static string Subscope(string address)
```

Comment:
I think we don't support `@` in the queues here because they need to be local. In that case I think I would use the `LogicalAddress` and `EndpointInstance` combo to generate the name. It is safer because when we later translate to transport address the transport-allowed delimiter is used instead of hardcoding to `.`

Reply:
@SzymonPobiega I'm not sure what you mean because there is no built-in translation that is happening. If I understood correctly we take those setting values that are written by SCMU or modified by the user and use it as is. I double checked and the v5 subscope logic always applied a dot to the queue so this would be still consistent for now.

Reply:
In V5 there was not translation. It has been introduced in V6. Here's the problem:
 * If we treat the values stored by SCMU/user as *transport address* then we allow various funny characters there i.e. SQL transport can use multiple `@` e.g. `myQueue@mySchema@myCatalog`. It would be impossible to figure out, for any transport, where to "inject" the subscope.
 * If we treat the values stored by SMU as *logical endpoint name* then we can use the `Subscope` in `LogicalAddress` and then employ the transport to translate. The downside is that you can't connect SC to any given non-standard catalog (in SQL transport)

I think that I like the second option more as parsing the transport-specific address string is dangerous.

Reply:
We looked at this and we figured out that although we would like to fix it, this seems to be a very deep rabbit hole. We will address this in a separate PR. We discussed following points:
 * The core expects the input queue in the logical form (endpoint name)
 * The core expects any output queues in the physical form (address)
 * We could apply the same logic in SC/SCMU
   * audit/error queues are provided as *logical* adresses
   * audit log/error log queues are provided as *physical* addresses
 * In case the user does not provide the *log queues we generate the physical address from the logical address or audit/error queue
 * Open question: should we apply the distribution strategies for the audit/error log